quamina.go

```diff
@@ -101,7 +99,7 @@ func New(opts ...Option) (*Quamina, error) {
 			return nil, err
 		}
 	}
-	if !(q.mediaTypeSpecified || q.flattenerSpecified) {
+	if !q.mediaTypeSpecified && q.flattener == nil {
```

Comment:
Could we please fully parenthesize this kind of condition?  As a matter of principle I never remember the precedence rules in any programming languages, which forces me to put in a few unnecessary parentheses, which I suspect has saved me from disasters more than once. A
```go
if (!q.mediaTypeSpecified) && (q.flattener == nil)

Reply:
done